File: gitrepo/python/liczby23.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-
#
#  liczby23.py
  
def liczby2():
    """
    funkcja drukuje wszystkie liczby 2-cyfrowe, w których 
    nie powtarzają się cyfry. Na koniec zwraca ilość takich liczb.
    wykluczone liczby: 11,22,33
    """
    
    ile = 0 #licznik liczb
    for i in range(1, 10): #pętla dziesiątek
        for j in range(10): #pętla jdności 
            if i != j:
                print("{}{} ".format(i, j), end=" ")
                ile = ile + 1 # zliczenie liczb
                    
    return ile
    
    
def liczby3():
    """
    funkcja drukuje wszystkie liczby 3-cyfrowe, w których 
    nie powtarzają się cyfry. Na koniec zwraca ilość takich liczb.
    wykluczone liczby: 110,222,333,334
    """
    ile = 0
    for i in range(1, 10): 
        for j in range(10):
            for k in range(10):
                if i != j != k != i:
                    print("{}{}{} ".format(i, j, k), end=" ")
                    ile = ile + 1 
    return ile
                    
def main(args):
    print("\n\nLiczb 2-cyfrowych:",liczby2())
    print(" ")
    print(" ")
    print("\n\nLiczb 3-cyfrowych:",liczby3())
    return 0

if __name__ == '__main__':
    import sys
    sys.exit(main(sys.argv))

File: gitrepo/cpp/liczby23.cpp
/*
 * liczby23.cpp 
 */


#include <iostream>
using namespace std;
int liczby2() {
    
    int ile = 0; //deklaracja
    for (int i = 1; i < 10; i++) {
        for (int j = 1; j < 10; j++) {
            if (i != j) {
                cout << i << j << " ";
                ile++;
            }
        
        }
    
    }
    return ile;
} 

int liczby3() {
    int ile = 0;
    for (int i = 1; i < 10; i++) {
        for (int j = 1; j < 10; j++) {
            for (int k = 1; k < 10; k++){
                if (i != j && i != k && k != j){
                    cout << i << j << k <<" ";
                    ile++;
                }
            }
        }
    }
    return ile;
}

int main(int argc, char **argv)
{
	cout << "\n\nLiczb 2-cyfrowych: " << liczby2();
    cout << "\n\nLiczb 3-cyfrowych: " << liczby3();
    
	return 0;
}

